packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -121,6 +123,73 @@ describe('ComboBox', function () {
     expect(label).toBeVisible();
   });
 
+  it('can be disabled', function () {
+    let {getByRole} = renderComboBox({isDisabled: true});
+
+    let combobox = getByRole('combobox');
+    act(() => {
+      combobox.focus();
+      userEvent.type(combobox, 'One');
+      jest.runAllTimers();
+    });
+
+    expect(() => getByRole('listbox')).toThrow();
+    expect(onOpenChange).not.toHaveBeenCalled();
+    expect(onFocus).not.toHaveBeenCalled();
+
+    act(() => {
```

Comment:
From trials in other places, I think act's need to be wrapped around each thing that changes state. that way everything updates, then the next action is taken

Reply:
same thing above on 131

Reply:
not entirely sure what needs to be done here, is the current wrapping `act` insufficient?

Reply:
ah are you saying the focus and type need to be in their own act?

Reply:
yeah, each thing that results in state changes should be in its own act from what i can tell, happy to be corrected though if that's not the case

Reply:
I'm not sure myself either, `act` is still a bit mysterious to me. All I understand about it is that you use it to prepare your test component for assertions (aka render, clicks, stuff that changes the component state goes into an `act`). I dunno how much state/render change you can bundle into a single `act` but there is an example [here](https://reactjs.org/docs/testing-recipes.html#events) that has 5 clicks happen in a single `act`

Reply:
ugh... then i don't know what the issue was when i tried to bundle them up as well, but i know it resulted in me splitting it in various places https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/test-utils/src/triggerPress.js =/
happy to go with them combined unless there's a problem though